lib/WUI/netdev.c

```diff
@@ -222,22 +222,34 @@ netdev_status_t netdev_get_status(uint32_t netdev_id) {
     }
 }
 
-netdev_ip_obtained_t netdev_get_ip_obtained_type() {
-    return IS_LAN_STATIC(ETH_CONFIG().lan.flag) ? NETDEV_STATIC : NETDEV_DHCP;
+netdev_ip_obtained_t netdev_get_ip_obtained_type(uint32_t netdev_id) {
+    return IS_LAN_STATIC(wui_netdev_config[netdev_id].lan.flag) ? NETDEV_STATIC : NETDEV_DHCP;
 }
 
 uint32_t netdev_set_dhcp(uint32_t netdev_id) {
+    ETH_config_t *pConfig = NULL;
+    err_t res = ERR_OK;
+
     if (netdev_id == NETDEV_ETH_ID) {
-        err_t res;
         res = netifapi_dhcp_start(&eth0);
-        CHANGE_FLAG_TO_DHCP(wui_netdev_config[netdev_id].lan.flag);
-        wui_netdev_config[netdev_id].var_mask = ETHVAR_MSK(ETHVAR_LAN_FLAGS);
-        save_eth_params(&wui_netdev_config[netdev_id]);
-        wui_netdev_config[netdev_id].var_mask = 0;
-        return res;
+        pConfig = &wui_netdev_config[netdev_id];
+    } else if (netdev_id == NETDEV_ESP_ID) {
+        esp_sta_quit(NULL, NULL, 1);
+        esp_sta_join(ap.ssid, ap.pass, NULL, 0, NULL, NULL, 0);
```

Comment:
Could you add a comment why we do this quit-join dance? I know it now (to reset it so it gives up potential static IP), but a future reader few years from now will rightfully wonder what might be happening.